tests/e2e/customsuites/directorypermissions.go

```diff
@@ -0,0 +1,938 @@
+	// they must be consistent or applications could malfunction.
+	ginkgo.It("should report consistent permissions via access() and stat() syscalls", func(ctx context.Context) {
+		// Set up volume + pod as before...
+		res := createVolume(ctx, testRegistry.config, pattern,
+			DefaultNonRootUser, DefaultNonRootGroup, "0700")
+		pod, err := CreatePodWithVolumeAndSecurity(ctx, testFramework, res.Pvc, "",
+			DefaultNonRootUser, DefaultNonRootGroup)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer e2epod.DeletePodWithWait(ctx, testFramework.ClientSet, pod)
+
+		dir := "/mnt/volume1/permsdir"
+
+		// Create the directory
+		CreateDirInPod(testFramework, pod, dir)
+
+		// Get mode using stat()
+		statMode, _, _ := e2evolume.PodExec(testFramework, pod, fmt.Sprintf("stat -c '%%a' %s", dir))
+
+		// Now test access() permissions: check readability (R_OK), writability (W_OK), executability (X_OK)
+		_, _, errR := e2evolume.PodExec(testFramework, pod, fmt.Sprintf("sh -c 'test -r %s'", dir))
+		_, _, errW := e2evolume.PodExec(testFramework, pod, fmt.Sprintf("sh -c 'test -w %s'", dir))
+		_, _, errX := e2evolume.PodExec(testFramework, pod, fmt.Sprintf("sh -c 'test -x %s'", dir))
+
+		// Convert the numeric mode into the expected permission bits
+		// For 0700, expect R/W/X succeed for owner; fail for others
+		expectedMode := strings.TrimSpace(statMode)
+		gomega.Expect(expectedMode).To(gomega.Equal("700"))
+
+		// Confirm access() matches: we expect R/W/X to succeed
+		gomega.Expect(errR).NotTo(gomega.HaveOccurred(), "access(R_OK) failed unexpectedly")
+		gomega.Expect(errW).NotTo(gomega.HaveOccurred(), "access(W_OK) failed unexpectedly")
+		gomega.Expect(errX).NotTo(gomega.HaveOccurred(), "access(X_OK) failed unexpectedly")
+	})
+
+	// --------------------------------------------------------------------
+	// 16. Directory Name Edge Cases (Full Path Test)
+	//
+	// This test ensures that Mountpoint-S3 can handle directories with
+	// complex names (Unicode, long names, special characters) and that
+	// full read/write functionality inside these directories works.
+	//
+	// Why? S3 key encoding applies to directory paths just like files,
+	// and we must ensure the driver isn't mishandling edge cases.
+	//
+	// Test steps for each case:
+	//
+	// ┌─────────────────────────────────────────────────────────────┐
+	// │ 1. mkdir <dir>                                              │
+	// │ 2. stat <dir>                                               │
+	// │ 3. touch <dir>/.marker  (store 'ok' text in the file)       │
+	// │ 4. cat <dir>/.marker    (read it back to confirm)           │
+	// │                                                             │
+	// │ Directory cases:                                            │
+	// │  - unicode-🚀🌟                                             │
+	// │  - long-(200 chars)                                         │
+	// │  - special-!@#$%^&()[]{}=+,~                                │
```

Comment:
Should you test as well non-printable (we don't accept them right now it returns an error)

And path traversal like `../../../` what's the behavior on this ?

Reply:
Actually unicode tests do not belong here, I removed them
We rely on mountpoint-s3 for now for behaviors of object keys.

I added the tests for path traversal, I expect the same permissions.